cheatsheets/Vulnerable_Dependency_Management_Cheat_Sheet.md

```diff
@@ -0,0 +1,233 @@
+
+To obtain these informations, we use the CVE content to know which kind of vulnerability affect the dependency. The `description` property give the answer: SQL injection, Remote Code Execution, Cross-Site Scripting, Cross-Site Request Forgery...
+
+Now, we know which kind of patching we must perform (it's become similar to [Case 2](#case-2) with the protection code) and where to add it.
+
+*Example:*
+
+We have an application using the Jackson API in a version exposed to the [CVE-2016-3720](https://nvd.nist.gov/vuln/detail/CVE-2016-3720).
+
+The description of the CVE is:
+
+```text
+XML external entity (XXE) vulnerability in XmlMapper in the Data format extension for Jackson 
+(aka jackson-dataformat-xml) allows attackers to have unspecified impact via unknown vectors.  
+```
+
+Based on these information, we determine that the patching will be to add a [pre-validation of any XML data](XML_External_Entity_Prevention_Cheat_Sheet.md) passed to the Jakson API to prevent [XML external entity (XXE)](https://www.acunetix.com/blog/articles/xml-external-entity-xxe-vulnerabilities/) vulnerability.
+
+**Step 3:**
+
+If possible, create a unit test trying to trigger the vulnerability via your application in order to ensure that the patch is effective and have a way to continuously ensure that the patch is in place during the evolution of the project.
+
+If you have a set of automated unit or integration or functionnal or security tests that exists for the application then run them to verify that the patch do not impact the stability of the application.
+
+## Case 4
+
+### Context
+
+The vulnerable dependency in found via the discovery of a full disclosure post on Internet, provider not aware of the vulnerability.
+
+### Ideal condition of application of the approach
+
+Provider collaborate with you after being notified of the vulnerability.
+
+### Approach
+
+**Step 1:**
+
+Inform the provider about the vulnerability by sharing the post with him.
+
+**Step 2:**
+
+Using the information of the full disclosure post, if the provider collaborate then apply the [Case 2](#case-2) otherwise apply the [Case 3](#case-3) but replace CVE information by the information of the full disclosure post.
+
+# Tools
+
+This section list several tools that can used to analyse the dependencies used by a project in order to detect the vulnerabilities.
+
+* Free
+    * [OWASP Dependency Check](https://www.owasp.org/index.php/OWASP_Dependency_Check): 
+        * Full support: Java, .Net.
+        * Experimental support: Python, Ruby, PHP (composer), NodeJS, C, C++.
+    * [NPM Audit](https://docs.npmjs.com/cli/audit)
+        * Full support: NodeJS, Javascript.
+        * HTML report available via this [module](https://www.npmjs.com/package/npm-audit-html).
+    * To manage the vulnerable dependencies in a transversal way in a organization the [OWASP Dependency Track](https://dependencytrack.org/) project can be used.
+* Commercial
```

Comment:
We can also suggest https://renovatebot.com - this tools does not find vulnerabilities but helps with updating all dependencies to the newest versions

Reply:
Added in coming commit as well as https://requires.io/